TrigScint/include/TrigScint/Firmware/trackproducer.h

```diff
@@ -0,0 +1,15 @@
+#ifndef TRACKPRODUCER_H
+#define TRACKPRODUCER_H
+
+#include "objdef.h"
+
+void copyCluster1(Cluster One, Cluster Two);
+void copyCluster2(Cluster One, Cluster Two);
+void trackproducer_ref(Cluster Pad1[NTRK], Cluster Pad2[NCLUS],
+                       Cluster Pad3[NCLUS], Track outTrk[NTRK],
+                       ap_int<12> lookup[NCENT][COMBO][2]);
+void trackproducer_hw(Cluster Pad1[NTRK], Cluster Pad2[NCLUS],
+                      Cluster Pad3[NCLUS], Track outTrk[NTRK],
+                      ap_int<12> lookup[NCENT][COMBO][2]);

```

Comment:
It feels weird that Pad1 would have length `NTRK` while the other `Cluster` arrays would have length `NCLUS`. Is this on purpose?

Reply:
Yes. The NTRK should be able to be smaller ultimately because we dont have the bandwidth in firmware to read out every track if there are more than 5 or so in a 25 ns clock cycle. The way the firmware handles this is that it handles the first cluster a little differently throughout, allowing it to read in a smaller number of clusters there. The number of tracks is less than or equal to the number of pad1 clusters and therefore the track max is inbuilt by a pad1 cluster max.

Reply:
I think its set to 10, which is way more clusters than we expect in the ldmx environment on a single layer but it is still handleable by our firmware. This processor will try to model the firmware precisely.

Reply:
That makes sense, can you add a comment explaining this so future developers don't think its a typo and replace it?